src/string-calculator.ts

```diff
@@ -1,12 +1,235 @@
+            return this.errorMessage;
+        }],
+        [ErrorCode.LastNumberMissedErrorCode, (): string => {
+            if (!this.isUnexpectedEOLAdded) {
+                if (this.errorMessage.trim().length != 0) {
+                    this.errorMessage = this.errorMessage.concat('\n');
+                }
+                this.errorMessage = this.errorMessage.concat(`Number expected but EOF found.`);
+                this.isUnexpectedEOLAdded = true;
+            }
+            this.lastErrorCode = ErrorCode.LastNumberMissedErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.NegativeNumberErrorCode, (number: string): string => {
+            if (this.lastErrorCode === ErrorCode.NegativeNumberErrorCode) {
+                this.errorMessage = this.errorMessage.concat(", " + number);
+            } else {
+                if (this.errorMessage.trim().length != 0) {
+                    this.errorMessage = this.errorMessage.concat('\n');
+                }
+                this.errorMessage = this.errorMessage.concat(`Negative not allowed : ` + number);
+            }
+            this.lastErrorCode = ErrorCode.NegativeNumberErrorCode;
+            return this.errorMessage;
+        }],
+        [ErrorCode.NumberNotNumericErrorCode, (number: string): string => {
+            let unexpectedChar = this.getUnexpectedChar(number);
+            let position = this.numberList.indexOf(unexpectedChar);
+
+            if (this.errorMessage.trim().length != 0) {
+                this.errorMessage = this.errorMessage.concat('\n');
+            }
+            this.errorMessage = this.errorMessage.concat(`'${this.customSep}' expected but '${unexpectedChar}' found at position ${position}.`)
+
+            this.lastErrorCode = ErrorCode.NumberNotNumericErrorCode;
+            return this.errorMessage;
+        }]
+    ]);
+
+    public summation(numberList: string): string {
+        this.numberList = numberList;
+
+        if (this.numberList.length === 0) {
             return "0";
         }
 
+        let result: number = 0;
+        let regexToSplitListIntoNumbers = this.createRegex();
+        let listOfNumbers: string[] = this.numberList.split(new RegExp(regexToSplitListIntoNumbers));
+
+        listOfNumbers.forEach(number => {
+
+            let err: ErrorCode = this.checkForErrors(number);
+            if(!err) {
+                result += Number(number);
+            }
+            else {
+                this.errorMessage = this.map.get(err);
+            }
+
+        });
+
+
+        if (this.errorMessage) {
+            return this.errorMessage;
+        }
+        return result.toString();
+    }
+
+    private createRegex(): string {
+        let res = ',|\n';
+        const _ = require('lodash');
+        if (this.numberList.includes("//")) {
+            let sep: string = this.getCustomSeparator();
+            res = _.escapeRegExp(sep);
+            this.customSep = sep;
```

Comment:
so we have a separator property but don't use it for parsing? sounds strange right?